Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -3112,6 +3168,39 @@ public void run() {
 		});
 	}
 
+	private void processRedeemRewardsResponse(final ServerResponse resp) {
+		boolean updateListener = false;
+		JSONObject requestObject = resp.getRequestObject();
+		if (requestObject != null) {
+			if (requestObject.has("bucket") && requestObject.has("amount")) {
+				try {
+					int redeemedCredits = requestObject.getInt("amount");
+					String creditBucket = requestObject.getString("bucket");
+					updateListener = redeemedCredits > 0;
+
+					int updatedCreditCount = prefHelper_.getCreditCount(creditBucket) - redeemedCredits;
+					updatedCreditCount = updatedCreditCount > 0 ? updatedCreditCount : 0;
```

Comment:
Is it possible for this to be less than zero? Also, why not just like `Math.max(count, 0)`?